src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -395,11 +235,48 @@ public virtual bool ShouldConvertChallengeToForbidden()
         /// deals an authentication interaction as part of it's request flow. (like adding a response header, or
         /// changing the 401 result to 302 of a login page or external sign-in location.)
         /// </summary>
-        /// <returns></returns>
-        protected virtual Task ApplyResponseChallengeAsync()
+        /// <param name="context"></param>
+        /// <returns>True if no other handlers should be called</returns>
+        protected virtual Task<bool> HandleUnauthorizedAsync(ChallengeContext context)
         {
-            ApplyResponseChallenge();
-            return Task.FromResult(0);
+            Response.StatusCode = 401;
```

Comment:
Should we just flip these to true to block the chaining?  We would need to set 401 explicitly in all of the children if we don't do it here.  This is how we get the default 401/403 behavior if you don't override these methods